hooks/src/index.d.ts

```diff
@@ -74,7 +74,7 @@ type CreateHandle = () => object;
  * ref.current
  * @param inputs If present, effect will only activate if the values in the list change (using ===).
  */
-export function useImperativeHandle<T>(ref: Ref<T>, create: CreateHandle, inputs?: Inputs): void;
+export function useImperativeHandle<T>(ref: PreactRef<T>, create: CreateHandle, inputs?: Inputs): void;
```

Comment:
as mentioned above, `CreateHandle` should probably also have the type param, could copy from the react typings

```suggestion
export function useImperativeHandle<T, R extends T>(ref: PreactRef<T>, create: () => R, inputs?: Inputs): void;
```

Reply:
also have to remove the type above though

sometimes think maybe we should just extend the react types instead of maintaining them independently?

Reply:
That's something we can investigate once X is out :+1: